sdk/python/kfp/_local_client.py

```diff
@@ -0,0 +1,441 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import datetime
+import json
+import logging
+import re
+import subprocess
+from typing import Any, Callable, Dict, List, Mapping, Tuple, Union, cast
+
+import kfp
+
+
+class _Dag:
```

Comment:
Can we also add a docstring briefing what `_Dag` is for? Thanks

Reply:
Some docstring added, is it enough?

Reply:
Yep. This LGTM. Thanks